src/connection.c

```diff
@@ -230,13 +227,17 @@ static void handleResolved(void *arg)
 
     conn->rc = redisAsyncConnectWithOptions(&options);
     if (conn->rc->err) {
-        conn->state = CONN_CONNECT_ERROR;
-        conn->connect_errors++;
+        goto fail;
+    }
 
-        redisAsyncFree(conn->rc);
-        conn->rc = NULL;
-        return;
+#ifdef HAVE_TLS
+    if (conn->rr->config->tls_enabled) {
+        if (redisInitiateSSLWithContext(&conn->rc->c, conn->rr->ssl) != REDIS_OK) {
+            CONN_LOG_ERROR(conn, "SSL Error!");
```

Comment:
TLS issues are hard to debug, we need to make sure we provide more information here from OpenSSL.